File: Arrays/Merge Intervals/code.java
class Solution {
    public int[][] merge(int[][] intervals) {
        if(intervals ==null || intervals.length<=1){
            return intervals;
        }
        Arrays.sort(intervals, (a,b)->Integer.compare(a[0],b[0]));
        List<int[]> mergedIntervals=new ArrayList<>();
        int[] currentInterval=intervals[0];
        for(int i=1;i<intervals.length;i++){
            int[] interval=intervals[i];
            if(currentInterval[1]>=interval[0]){
                currentInterval[1]=Math.max(currentInterval[1],interval[1]);
            }else{
                mergedIntervals.add(currentInterval);
                currentInterval=interval;

            }

        }
        mergedIntervals.add(currentInterval);
        int[][] result =new int[mergedIntervals.size()][2];
        for(int i=0;i<mergedIntervals.size();i++){
            result[i]=mergedIntervals.get(i);
        }
        return result;

        
    }
}

File: Arrays/Merge Intervals/code.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        sorted_intervals=sorted(intervals, key=lambda x: x[0])
        print(intervals)
        ans=[]
        current_list=sorted_intervals[0]
        for i in sorted_intervals:
            temp=i
            if(current_list[1]>=temp[0]):
                current_list[1]=max(current_list[1],temp[1])
            else:
                ans.append(current_list)
                current_list=temp
        ans.append(current_list)
        return ans
                

        
